velox/exec/HashBuild.cpp

```diff
@@ -377,6 +387,31 @@ void HashBuild::addInput(RowVectorPtr input) {
     return;
   }
 
+  if (dropDuplicates_ && !abandonBuildNoDupHash_) {
+    const bool abandonEarly = abandonBuildNoDupHashEarly(table_->numDistinct());
```

Comment:
Curious why we abandon? In what scenarios the dedup way is less memory/cpu efficient? If for all general cases it is more efficient, should we always make this special case (dedup leftsemi and anti) without abandoning?